File: src/Kattis/COMP321/Contests/Contest1/M.py
c, s = list(map(int, input().split()))
cp = 0
possible = True
for _ in range(s):
    l, e, n = list(map(int, input().split()))
    cp -= l
    if not possible:
        continue
    if cp < 0:
        possible = False
    cp += e
    if cp > c:
        possible = False
    if cp < c and n > 0:
        possible = False

if cp != 0:
    possible = False
print("possible" if possible else "impossible")

File: src/Kattis/Programming Challenges/Contests/C1/M.cpp
#include <iostream>

using namespace std;

int main()
{
    int C, n, leave, enter, stay, num_passengers = 0;
    cin >> C >> n;
    bool possible = true;
    for (int i = 0; i < n; i++)
    {
        cin >> leave >> enter >> stay;
        num_passengers -= leave;
        if (num_passengers < 0)
        {
            possible = false;
            break;
        }
        num_passengers += enter;
        if (num_passengers > C)
        {
            possible = false;
            break;
        }
        if (num_passengers < C && stay > 0)
        {
            possible = false;
            break;
        }
    }
    if (num_passengers != 0)
    {
        possible = false;
    }
    if (possible)
    {
        cout << "possible" << endl;
    }
    else
    {
        cout << "impossible" << endl;
    }
    return 0;
}
